frontend/src/components/unsubscribe/tests/Unsubscribe.test.tsx

```diff
@@ -62,14 +70,16 @@ test('successfully unsubscribes user from a campaign', async () => {
   expect(heading).toBeInTheDocument()
 })
 
-test('displays thank you page when user does not unsubscribe', async () => {
+test('displays thank you page when user resubscribe', async () => {
   // Setup
   server.use(...mockApis())
   renderUnsubscribe()
 
+  await unsubscribe()
+
   // Wait for the component to fully load
   const stayButton = await screen.findByRole('button', {
-    name: /stay/i,
+    name: /subscribe me back/i,
   })
```

Comment:
Do we want to mock the error cases too? E.g. when the hash is wrong, or when the "Other" reason is too long (is there any limit on this?)

Reply:
Nope no restriction on length of Other reason